### ISSUE TITLE
Adding Tuplets To Linked Staff Corrupts Score File

### ISSUE THREAD

**Issue by MillisecondBleeper** [musescore/MuseScore#18879]
### Issue type

File corruption

### Bug description

When a score has a part with a linked staff open, adding a tuplet affects the primary staff in the part and main score and the linked staff in the focused part, but doesn't correctly add the tuplet mark to the linked staff in the other part, leaving it with too many notes and marking the file corrupt.

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore version (64-bit): 4.1.1-232071203, revision: [e4d1ddf](https://github.com/musescore/musescore/commit/e4d1ddf)

[Demonstration Video](https://youtu.be/HonO6o7irVo)    
[examples (before & after).zip](https://github.com/musescore/MuseScore/files/12234887/examples.before.after.zip)    
[MuseScore Forum Post](https://musescore.org/en/node/353108#comment-1201665)

### Steps to reproduce

1. Create New score with piano and guitar initially. Scope of issue is unclear, but this is my case.
1. Add a linked staff and set to tabulature for the guitar
2. Open the part score for the guitar
3. in either the main or part score, add a tuplet to the guitar on either staff
5. issue occurs, and attempting to save will warn user of corrupted score.

### Screenshots/Screen recordings

https://youtu.be/HonO6o7irVo

### MuseScore Version

4.1.1-232071203, revision: [e4d1ddf]

### Regression

I don't know

### Operating system

Windows 10 Pro 22H2

### Additional context

[save error](https://github.com/musescore/MuseScore/assets/67459516/5fb6f038-0302-4e63-b7b6-728230be90b8)

**Comment by speeves** [comment]
I can verify that this fixed my issues with tuplets: 
OS: Ubuntu 23.04, Arch.: x86_64, MuseScore version (64-bit): 4.2.0-232670304, revision: github-musescore-musescore-6d7d80b

**Comment by speeves** [comment]
To follow up once more. I have found this work-around, if you stick with the stable release:

 OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore version (64-bit): 4.1.1-232071203, revision: github-musescore-musescore-e4d1ddf

I found that I can reset the corrupted part and then save the score.

Here are the steps:

1. Enter triplet into score with guitar notation and linked tablature.
2. Click Save
3. Receive the corrupted score message.
4.  Click Parts > Electric Guitar > Meatball menu (horizontal dots) and Click Reset
5. Try to save your score again.
6. Success!!
